WordPress/gutenberg
# Fix query block duplicate group block rename controls for WP 6.4

## What?


Fixes the repeated `Rename` menu item in the block options menu when the Group block is within a Query Loop block. 

Fixes https://github.com/WordPress/gutenberg/issues/55517

## Why?


Because you only want to rename once, rename once, rename once, rename once...

## How?


Removes the reliance on `Block.Edit` hook and instead moves the menu item directly into the `BlockSettingsMenuControlsSlot`.

Why is this ok?

For 6.4 this feature exists only for the Group block. However in Gutenberg `trunk` [the feature is enabled for (nearly) all blocks](https://github.com/WordPress/gutenberg/pull/54426).

This means the feature is an integral part of the Editor and so using a hook no longer makes sense. Indeed, we agreed on such an approach in https://github.com/WordPress/gutenberg/pull/55250#issuecomment-1759928887.

## Testing Instructions





- Create ~lots of~ 31 Posts (I used Fakerpress)
- Add a Query loop block
- Add a Group block within the Post Template
- Add another Group block _outside_ of the Query loop.
- Click on the Group inside the Query Loop and trigger the block options menu in the list view. You should see only a _single_ `Rename` option.
- Do the same on the Group outside the loop.
- Validate that the renaming functionality otherwise behaves as per the `wp/6.4` branch.
- Also check that `Rename` is only available for Group
- Check that you can still rename the block via the inspector controls `Advanced -> Rename`.

### Testing Instructions for Keyboard


## Screenshots or screencast
